fedimint-api/src/bitcoin_rpc.rs

```diff
@@ -0,0 +1,147 @@
+use std::future::Future;
+use std::sync::Arc;
+use std::time::Duration;
+
+use async_trait::async_trait;
+use bitcoin::{Block, BlockHash, Network, Transaction};
+use fedimint_api::Feerate;
+use thiserror::Error;
+use tracing::info;
+
+use crate::dyn_newtype_define;
+
+#[derive(Error, Debug)]
+pub enum Error {
+    #[error("Rpc error")]
+    Rpc(#[from] anyhow::Error),
+}
+
+pub type Result<T> = std::result::Result<T, Error>;
+
+/// Trait that allows interacting with the Bitcoin blockchain
+///
+/// Functions may panic if if the bitcoind node is not reachable.
+#[async_trait]
+pub trait IBitcoindRpc: Send + Sync {
```

Comment:
Could we make this part of the new `fedimint-bitcoind` crate?

Reply:
Yes, though it would reverse the dependency arrow, where either `fedimint-server` or `fedimint-api` would now need to import `fedimint-bitcoind`. Seems undesirable. As it is RN it is akin to `Database` where `fedimint-rocksdb` is an inplementation crate for an interface from `fedimint-api`.

Reply:
Why do these need it? Tests? It's different from `Database` in that it's only used by the wallet module and some tests that have to mock `bitcoind`. It's also used by any fedimint instantiation that has a wallet module, but there the wallet module could just re-export the necessary types.

Reply:
OK. It is actually doable and done. I guess `BitcoindRpc` should not be there in the first place.

Reply:
Ah. Now I see the issue. Wasm32 can't compile that code, so the implementation has to be somewhere conditional, but the interface somewhere unconditional.

Reply:
I see, could we give `fedimint-bitcoind` a feature flag instead?